### PR TITLE
wootility: implement `NIXOS_OZONE_WL` and add `.desktop` file

### PR DESCRIPTION
**Pull request by sodiboo**
## Description of changes

Now, the icon and `.desktop` file for wootility are copied to the output. Additionally, we wrap the main executable with a `makeWraper` that adds ozone platform hints. 

With this change, i can both open wootility from my fuzzel (rather than the terminal) and it works fine on wayland without passing flags manually. I've tested this on my own system and it works great.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/tools/misc/wootility/default.nix [hunk 1]
```diff
@@ -32,7 +47,7 @@ appimageTools.wrapType2 rec {
     description = "Customization and management software for Wooting keyboards";
     platforms = [ "x86_64-linux" ];
     license = "unknown";
```
Comment (Aleksanaa) [inline pkgs/tools/misc/wootility/default.nix:49]:
```suggestion
    license = licenses.unfree;
```

File: pkgs/tools/misc/wootility/default.nix [hunk 2]
```diff
@@ -15,6 +16,20 @@ appimageTools.wrapType2 rec {
...
+    in ''
+      source "${makeWrapper}/nix-support/setup-hook"
+      wrapProgram $out/bin/wootility \
+        --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform-hint=auto --enable-features=WaylandWindowDecorations}}"
+
+      install -Dm444 ${contents}/wootility-lekker.desktop -t $out/share/applications
+      install -Dm444 ${contents}/wootility-lekker.png -t $out/share/pixmaps
+      substituteInPlace $out/share/applications/wootility-lekker.desktop \
+        --replace-fail 'Exec=AppRun' 'Exec=wootility' \
+        --replace-warn 'Name=wootility-lekker' 'Name=Wootility'
```
Comment (Aleksanaa) [inline pkgs/tools/misc/wootility/default.nix:30]: Why do we replace it?

File: pkgs/tools/misc/wootility/default.nix (hunk 1 above)
Comment (sodiboo) [reply pkgs/tools/misc/wootility/default.nix]: Good catch!

File: pkgs/tools/misc/wootility/default.nix (hunk 2 above)
Comment (sodiboo) [reply pkgs/tools/misc/wootility/default.nix:30]: This application is called `Wootility` in all user-facing places (including the nixpkgs pname); the name `wootility-lekker` is an internal codename referring to a specific branch of development (like the `firefox` package is packaging Firefox Quantum; but that's not the user-facing name). See e.g. their website (https://wooting.io/wootility), and names the app uses to reference itself (screenshot attached of titlebar and settings page)

image

Additionally, it is stylized with a capital first letter, which is typical for desktop entries, and it matches Wooting's branding (see screenshot, especially titlebar)

**Pull request merged**